docs/ref/templates/builtins.txt

```diff
@@ -959,66 +959,100 @@ output (as a string) inside a variable. This is useful if you want to use
 
 .. versionadded:: 5.1
 
-Outputs the query string from a given :class:`~django.http.QueryDict` instance,
-if provided, or ``request.GET`` if not and the
-``django.template.context_processors.request`` context processor is enabled.
-If the ``QueryDict`` is empty, then the output will be an empty string.
-Otherwise, the query string will be returned with a leading ``"?"``.
+Outputs a URL-encoded formatted query string based on the provided parameters.
 
-If not using the ``django.template.context_processors.request`` context
-processor, you must pass either the ``request`` into the template context or a
-``QueryDict`` instance into this tag.
+This tag requires a :class:`~django.http.QueryDict` instance, which defaults to
+:attr:`request.GET <django.http.HttpRequest.GET>` if none is provided, to
+generate the resulting query string from the passed key-value pairs.
```

Comment:
```suggestion
:attr:`request.GET <django.http.HttpRequest.GET>` if none is provided.
```
I think less is more here as we don't have to pass in key-value pairs explicitly and I think this could be confusing

Reply:
Agreed, removing!